### PR TITLE
Add RGBA channels to detect the monochromatic

### PR DESCRIPTION
**Pull request by aurangzaib048**
Edge cases:
```
domains = [
        "https://www.theguardian.com/",
        "https://newyorker.com",
        "https://www.edmunds.com/",
        "https://www.businessinsider.com/",
        "https://forbes.com",
        "https://wired.com",
        "https://www.pocket-lint.com/",
        "https://pitchfork.com/",
        "https://www.cbr.com/",
        "https://cnet.com",
        "https://www.usatoday.com/",
        "https://www.fool.com/",
        "https://animenewsnetwork.com",
        "https://www.bbc.com/",
        "https://aljazeera.com",
        "https://gamerescape.com",
        "https://www.spiceworks.com/",
        "https://gsmarena.com",
    ]
```
Monochromatic Algo: https://github.com/brave/news-aggregator/pull/649/files#diff-1c5d796c1fafb2c33717ea2bbb01bb3bd7e166d8f20250f8fd88926d9e72be8aR52

### CODE REVIEW

File: src/favicons_covers/color.py [hunk 1]
```diff
@@ -28,25 +28,43 @@ def fade_to_brighter_color(hex_color, fade_factor=0.12):
...
+#         return False
+
+
+def is_monochromatic(image, tolerance=25):
+    colored_image = image.convert("RGBA")
+
+    image_array = np.array(colored_image)
+
+    std_rgb = [
+        int(np.std(image_array[:, :, 0][image_array[:, :, 0] != 0])),
```
Comment (LorenzoMinto) [inline src/favicons_covers/color.py:58]: This is filtering on lack of signal on that same channel, not sure what was the intention here. We should filter out values where `alpha = 0` 
```
np.std(image_array[:, :, 0][image_array[:, :, 3] != 0]
```
Same for below.

File: src/favicons_covers/color.py [hunk 2]
```diff
@@ -28,25 +28,46 @@ def fade_to_brighter_color(hex_color, fade_factor=0.12):
...
+        int(np.std(image_array[:, :, 0][image_array[:, :, 3] != 0])),
+        int(np.std(image_array[:, :, 1][image_array[:, :, 3] != 0])),
+        int(np.std(image_array[:, :, 2][image_array[:, :, 3] != 0])),
+    ]
+
+    count_smaller_than_threshold = len(
+        [value for value in std_rgb if value < tolerance]
+    )
+
+    return count_smaller_than_threshold >= 1
```
Comment (LorenzoMinto) [inline src/favicons_covers/color.py:70]: This looks off. It's basically saying that if at least one channel has low variance then it is monochromatic, which may not be a sufficient condition.

Comment (aurangzaib048) [reply src/favicons_covers/color.py:70]: Yeah, true but I just wanted to see the results, we will iterate as per the results. As well as I wanted to see the results of the newly added API.

**Comment by aurangzaib048** [comment]
Yeah, that’s great idea, I have created an issue for this: https://github.com/brave/news-aggregator/issues/652

**Pull request merged**